docs/writing-rules/rule-syntax.md

```diff
@@ -759,6 +760,63 @@ int("2147483646")
 
 This removes quotes (`'`, `"`, and `` ` ``) from both ends of the metavariable content. As a result, Semgrep detects `"2147483648"`, but it does **not** detect `"2147483646"`. This is useful when you expect strings to contain integer or float data.
 
+### `metavariable-name`
+
+The `metavariable-name` operator adds a constraint to the types of identifiers a metavariable is able to match. Currently the only constraint supported is on module or namespace an identifier originates from. This is useful for filtering results in languages which don't have a native syntax for fully qualified names, or languages where module names may contain characters which are not legal in identifiers (e.g., JavaScript, TypeScript). 
+
+:::info
+The `metavariable-name` operator requires the Pro engine, since it requires additional name resolution information. Moreover, name resolution information is higher fidelity with interfile enabled, and thus `metavariable-name` will be more accurate with interfile naming.
+:::
+
+```yaml
+rules:
+  - id: insecure-method
+    patterns:
+      - pattern: $MODULE.method(...)
+      - metavariable-name:
+          metavariable: $MODULE
+          module: "@foo-bar"
+    message: Uses insecure method from @foo-bar.
+    languages:
+      - javascript
+    severity: ERROR
+```
+
+The pattern immediately above matches the following:
+
+```javascript
+import * as lib from '@foo-bar';
```

Comment:
I would show both import and require in the example, just to show that it works for both
```javascript
// ECMAScript modules
import * as lib from '@foo-bar'
import * as lib2 from 'myotherlib'

// CommonJS modules
const { method } = require('@foo-bar')
const lib3 = require('myotherlib')

// ruleid: insecure-method
lib.method("test")

// ruleid: insecure-method
method("test")

// ok: insecure-method
lib.method2("test")

// ok: insecure-method
lib2.method("test")

// ok: insecure-method
lib3.method("test")
```